src/Symfony/Component/Security/Core/Exception/CustomAuthenticationException.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * An authentication exception where you can add a message that will
+ * be shown to the user, without needing to create a sub-class.
+ *
+ *  throw CustomAuthenticationException::createWithSafeMessage(
+ *      'That was a ridiculous username'
+ *  );
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class CustomAuthenticationException extends AuthenticationException
```

Comment:
I would convey the "safe" (not sure if safe is the right word here) notion in the class name.